### PR TITLE
rgw: Modification in the usage of force-branch

### PR DESCRIPTION
**Pull request by TRYTOBE8TME**
Signed-off-by: Kalpesh Pandya <kapandya@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by yuvalif** [comment]
please fix the `make checl` error:
```
./tasks/notification_tests.py:15:1: F401 'teuthology.config.config as teuth_config' imported but unused
```

File: qa/tasks/notification_tests.py [hunk 1]
```diff
@@ -12,47 +12,28 @@
...
-    Download the bucket notification tests from the git builder.
-    Remove downloaded test file upon exit.
-    The context passed in should be identical to the context
-    passed in to the main task.
-    """
     assert isinstance(config, dict)
-    log.info('Downloading bucket-notification-tests...')
+    log.info('Downloading bucket-notifications-tests...')
     testdir = teuthology.get_testdir(ctx)
+    branch = ctx.config.get('suite_branch', None)
```
Comment (yuvalif) [inline qa/tasks/notification_tests.py:25]: what if you don't explicitly provide these values?
does it work in teuthology without them by taking the defaults?

File: qa/tasks/notification_tests.py [hunk 2]
```diff
@@ -12,47 +12,28 @@
...
         ctx.cluster.only(client).run(
-            args=[
-                'git', 'clone',
-                '-b', bntests_branch,
-                git_remote + 'ceph.git',
-                '{tdir}/ceph'.format(tdir=testdir),
-                ],
+            args=['git', 'clone', '-b', branch, repo + '.git', '{tdir}/ceph'.format(tdir=testdir)],
             )
-        if sha1 is not None:
```
Comment (yuvalif) [inline qa/tasks/notification_tests.py:48]: you can get sha1 from the commandline as well

File: qa/tasks/kafka.py [hunk 3]
```diff
@@ -173,6 +173,7 @@ def task(ctx,config):
     tasks:
     - kafka:
         client.0:
+          kafka_version: 2.6.0
```
Comment (yuvalif) [inline qa/tasks/kafka.py:176]: why do you need to specif the kafka version?

File: qa/tasks/notification_tests.py (hunk 1 above)
Comment (TRYTOBE8TME) [reply qa/tasks/notification_tests.py:25]: I was about to ask this question but I think I've a way to solve this which I'll try it out.

File: qa/tasks/notification_tests.py (hunk 2 above)
Comment (TRYTOBE8TME) [reply qa/tasks/notification_tests.py:48]: Okay, so I'll keep this part.

File: qa/tasks/kafka.py (hunk 3 above)
Comment (TRYTOBE8TME) [reply qa/tasks/kafka.py:176]: As per [https://github.com/ceph/ceph/pull/39139#discussion_r594554362](this) we added this parameter which I forgot to add as a reference so just added it here.

**Comment by TRYTOBE8TME** [comment]
Passing teuthology: https://pulpito.ceph.com/kapandya-2021-06-25_15:50:35-rgw:notifications-wip-rgw-notifications-tests-distro-basic-smithi/

File: qa/tasks/notification_tests.py [hunk 4]
```diff
@@ -17,34 +17,29 @@
...
-    Remove downloaded test file upon exit.
-    The context passed in should be identical to the context
-    passed in to the main task.
-    """
     assert isinstance(config, dict)
-    log.info('Downloading bucket-notification-tests...')
+    log.info('Downloading bucket-notifications-tests...')
     testdir = teuthology.get_testdir(ctx)
+    actual_branch = 'master'
+    branch = ctx.config.get('suite_branch', actual_branch)
```
Comment (alimaredia) [inline qa/tasks/notification_tests.py:27]: @kapandya could you add an example in the documentation in this file above task.py with `suite_repo` and `suite_branch`.

**Comment by TRYTOBE8TME** [comment]
Passing teutholofy for both the cases:

Case 1: When we **pass `suite_branch` and `suite_repo`** in the teuthology-suite command -> https://pulpito.ceph.com/kapandya-2021-07-01_10:15:17-rgw:notifications-wip-rgw-notification-tests-distro-basic-smithi/

Case 2: When we **don't pass either** of these parameters explicitly -> https://pulpito.ceph.com/kapandya-2021-07-01_10:15:45-rgw:notifications-wip-rgw-notification-tests-distro-basic-smithi/

**Review by yuvalif — APPROVED** [review]

**Review by alimaredia — APPROVED** [review]

**Pull request merged**